docs/index.bs

```diff
@@ -3112,7 +3112,19 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. If |request|'s [=request/destination=] is either "<code>embed</code>" or "<code>object</code>", then:
           1. Return null.
       1. Else if |registration|'s <a>active worker</a>'s [=service worker/list of router rules=] is [=list/is not empty=]:
-          1. If running [=Get Router Source=] algorithm with |registration|'s <a>active worker</a> and |request| returns "network", return null.
+          1. Let |source| be a result of running [=Get Router Source=] algorithm with |registration|'s <a>active worker</a> and |request|.
+          1. If |source| is {{RouterSource/"network"}}, return null.
+          1. Else if |source| is {{RouterSource/"cache"}}, then:
+              1. Let |requestResponses| be the result of running [=Query Cache=] with |request|.
+              1. If |requestResponses| is an empty [=list=], return null.
+              1. Else:
+                  1. Let |requestResponse| be the first element of |requestResponses|.
+                  1. Let |response| be a copy of |requestResponse|'s response.
+                  1. Let |settingsObject| be a new [=environment settings object=].
```

Comment:
Creating a new environment settings object doesn't seem correct. That represents a new Window or WorkerGlobalScope, but you haven't created one of those at the same time.

Since the purpose of this seems to be some sort of COOP check, shouldn't you be using the service worker's settings object?

Or... hmm, is the problem here that the ServiceWorkerGlobalScope might not be created at all yet? That seems hard to work around. So many algorithms, e.g. https://fetch.spec.whatwg.org/#cross-origin-resource-policy-check , seem to assume that you'll have the actual environment settings object. E.g. https://fetch.spec.whatwg.org/#queue-a-cross-origin-embedder-policy-corp-violation-report assumes that a global object exists.

How does this work in the implementation? Do we still send reports somehow?

Reply:
It is actually implemented like: https://source.chromium.org/chromium/chromium/src/+/main:content/browser/service_worker/service_worker_cache_storage_matcher.cc;l=49;drc=c686e8f4fd379312469fe018f5c390e9c8f20d0d
It gets COEP from `ServiceWorkerVersion`.  Since I did not have any ideas where to plumb the report, it actually do not.

Reply:
That seems problematic... so if you start using service worker static routing, you lose COOP reporting? I think some of our partners are using COOP reporting heavily.

Reply:
Ah, I did not know that.  Since half of the code calling AddReceiver sets std::NullRemote() as COEP reporter, I did not thought that important.
https://source.chromium.org/chromium/chromium/src/+/main:components/services/storage/public/mojom/cache_storage_control.mojom;l=43;drc=3cc22c22d7637bc5604e8fef3b0882b51a762901

To be clear, for the cache lookup, COEP violation may not be reported.  Other cases won't be affected.

Reply:
Hmm. I think it is worth starting a discussion with the teams that own COEP maybe?

(Also, sorry for stating COOP. I guess we are only talking about COEP here, not COOP.)

Reply:
Upon the suggestion from the COEP folk, I made request's COEP is used instead of SW's.  Will you take another look at this?
Thank you.